simpleflow/step/workflow.py

```diff
@@ -0,0 +1,69 @@
+import os
+import copy
+from collections import defaultdict
+
+from .constants import STEP_ACTIVITY_PARAMS_DEFAULT
+from .submittable import Step
+from .tasks import GetStepsDoneTask
+from simpleflow import activity, settings, task
+
+
+class WorkflowStepMixin(object):
+
+    def get_step_bucket(self):
+        """
+        Return the S3 bucket where to store the steps files
+        """
+        return '/'.join((settings.SIMPLEFLOW_S3_HOST, settings.STEP_BUCKET))
+
+    def get_step_path_prefix(self):
+        """
+        Return the S3 bucket's path prefix where to store the steps files
+        """
+        return os.path.join(self.get_execution_context().get("workflowId", "default"), 'steps')
+
+    def get_step_activity_params(self):
+        """
+        Returns the params for GetStepsDoneTask and MarkStepAsDone activities
+        Will be merged with the default ones
+        """
+        return {}
+
+    def add_forced_steps(self, steps, reason=None):
+        """
+        Add steps to force
+        """
+        if not hasattr(self, 'steps_forced'):
+            self.steps_forced = set()
+            self.steps_forced_reasons = defaultdict(set)
+        steps = set(steps)
+        self.steps_forced |= set(steps)
+        if reason:
+            for step in steps:
+                self.steps_forced_reasons[step].add(reason)
+
+    def get_forced_steps(self):
+        return list(getattr(self, 'steps_forced', []))
+
+    def _get_step_activity_params(self):
```

Comment:
We have `get_step_activity_params` which is public, and `_get_step_activity_params` which is the merged one. 
I added doc